dwave/cloud/client/base.py

```diff
@@ -725,22 +679,23 @@ def get_regions(self, refresh: bool = False) -> Dict[str, Dict[str, str]]:
 
         Returns:
             Mapping of region details (name and endpoint) over region codes.
+
+        .. deprecated:: 0.11.0
+            :meth:`.Client.get_regions` method is deprecated in favor of
+            :func:`dwave.cloud.regions.get_regions` function and will be
+            removed in 0.13.0.
+
         """
-        # make sure we don't cache by noise in config
-        config = self.config.model_copy(
-            update=dict(region=None, endpoint=None, token=None, client=None,
-                        solver=None, polling_schedule=None, polling_timeout=None),
-            deep=True)
-        try:
-            rs = Client._fetch_available_regions(config=config, refresh_=refresh)
-        except api.exceptions.RequestError as exc:
-            logger.debug("Metadata API unavailable", exc_info=True)
-            raise ValueError(
-                f"Metadata API unavailable at {self.config.metadata_api_endpoint!r}")
+        warnings.warn(
+            f"`Client.get_regions` method is deprecated since "
+            f"dwave-cloud-client 0.11.0, and will be removed in 0.13.0. "
+            f"Use `dwave.cloud.regions.get_regions` for greater flexibility instead.",
+            DeprecationWarning, stacklevel=2)
 
-        logger.info("Using region metadata: %r", rs)
+        rs = get_regions(config=self.config, refresh=refresh,
```

Comment:
perhaps `regions` instead of `rs`, it's only used once so nice to have a clear name

Reply:
Exactly because it has a short lifespan, this is fine. Also, this code will be removed in 0.13.0.